File: leaflet-view/src/main/java/com/gurunars/leaflet_view/NoPageRenderer.java
package com.gurunars.leaflet_view;

import android.view.View;

/**
 * Renderer to be used when there are no pages in the list.
 */
public interface NoPageRenderer {
    /**
     * Render a view to be shown when all pages are removed
     *
     * @return Rendered and populated view
     */
    View renderNoPage();

    /**
     * Called when the no page view is navigated into
     */
    void enter();
}

File: leaflet-view/src/main/kotlin/com/gurunars/leaflet_view/NoPageRenderer.kt
package com.gurunars.leaflet_view

import android.view.View

/**
 * Renderer to be used when there are no pages in the list.
 */
interface NoPageRenderer {
    /**
     * Render a view to be shown when all pages are removed
     *
     * @return Rendered and populated view
     */
    fun renderNoPage(): View

    /**
     * Called when the no page view is navigated into
     */
    fun enter()
}
